File: src/ProjectR/content/scripts/spells/WindStrike.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Wind Strike"
end

function GetDescription()
    return "Strike with the power of the wind."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .32
end

function GetDelay()
  return .65
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { WND }
end

function SpellEffect()
  local damage = (3.3 * aAD * (aWND/100) - dDEF) * (100/dWND)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/WindStrike.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class WindStrike : SpellScriptBase
    {
        public override string Name { get { return "Wind Strike"; } }
        public override string Description { get { return "Strike with the power of the wind."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.WND, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return .32; } }
        public override double Delay { get { return .65; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.3 * aAD * (aWND / 100) - dDEF) * (100 / dWND);
            DealDamage(damage);
        }
    }
}